application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -0,0 +1,351 @@
+     * <p>
+     * This fetches configuration information from a json configuration file (see
+     * {@link FreeCommandConfig}) for further details.
+     */
+    public FreeCommand() {
+        super(COMMAND_NAME, "marks this channel as free for another user to ask a question",
+                SlashCommandVisibility.GUILD);
+
+        channelIdToMessageIdForStatus = new HashMap<>();
+        channelMonitor = new ChannelMonitor();
+
+        isReady = false;
+    }
+
+    /**
+     * Reaction to the 'onReady' event. This method binds the configurables to the
+     * {@link net.dv8tion.jda.api.JDA} instance. Including fetching the names of the channels this
+     * command monitors.
+     * <p>
+     * It also updates the Status messages in their relevant channels, so that the message is
+     * up-to-date.
+     * <p>
+     * This also registers a new listener on the {@link net.dv8tion.jda.api.JDA}, this should be
+     * removed when the code base supports additional functionality
+     *
+     * @param event the event this method reacts to
+     */
+    @Override
+    public void onReady(@NotNull final ReadyEvent event) {
+        final JDA jda = event.getJDA();
+        // TODO remove this when onGuildMessageReceived has another access point
+        jda.addEventListener(this);
+
+        initChannelsToMonitor();
+        initStatusMessageChannels(jda);
+        logger.debug("Config loaded:\n{}", channelMonitor);
+
+        checkBusyStatusAllChannels(jda);
+        initStatusMessages(jda);
+
+        channelMonitor.statusIds()
+            .map(event.getJDA()::getTextChannelById)
+            .filter(Objects::nonNull)
+            .forEach(this::displayStatus);
+
+        isReady = true;
+    }
+
+    /**
+     * When triggered with {@code /free} this will mark a help channel as not busy (free for another
+     * person to use).
+     * <p>
+     * If this is called on from a channel that was not configured for monitoring (see
+     * {@link FreeCommandConfig}) the user will receive an ephemeral message stating such.
+     * 
+     * @param event the event that triggered this
+     */
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        logger.debug("/free used by {} on channel {}", event.getUser().getAsTag(),
+                event.getChannel().getName());
+        if (!handleShouldBeProcessed(event)) {
+            return;
+        }
+
+        long id = event.getChannel().getIdLong();
+        // do not need to test if key is present, shouldHandle(event) already does.
+        if (!channelMonitor.isChannelBusy(id)) {
+            FreeUtil.sendErrorMessage(event, UserStrings.ALREADY_FREE_ERROR.message());
+            return;
+        }
+        // TODO check if /free called by original author, if not put message asking if he approves
+        channelMonitor.setChannelFree(id);
+        // null impossible for event.getGuild (tested in handleShouldBeProcessed) ignore warning
```

Comment:
NIT: u _could_ replace this "long comment" by `Objects.requireNonNull(event.getGuild())` if u want.

Reply:
fixed